Week2.java

```diff
@@ -0,0 +1,98 @@
+package Homework;
+
+
+/*******************WEEK2****************************/
+
+/*Problem: triplets that sum to zero. 
+  TALK:
+Assuming unique elements in the array.
+taking size of the array as input.
+
+ EXAMPLES:
+ array of size=0, output: no triplet 
+ array of size<3 ([1,2]), output: no triplet  
+size= 5  [1,2,3,-1,-2],    output: -2 -1 3
+size=6   [5,-1,-4,-2,6],   output: -4 -1 5
+                                   -4 -2 6
+ size=4  [1,2,3,4],        output: no triplet                                   
+
+BRUTE FORCE:
+considering all the triplets.
+complexity: O(n3)
+
+OPTIMIZED SOLUTION AS BELOW 
+*/
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+//IMPLEMENT
+
+public class Week2 {
+    
+	 public static void TakeInput(int[] array, int size) {
+	    	if(size==0) {
+	    		return;
+	    	}
+	    	System.out.print("do");
+	    	Scanner s=new Scanner(System.in);
+	    	for(int i=0;i<size;i++) {
+	    		array[i]=s.nextInt();
+	    	}
+	    }
+	public static void SumOfTriplets(int []input, ArrayList<Variable> solution) {

```

Comment:
It is not typically common in Java (and most other modern managed languages) to pass in an empty list and expect it to be populated, that is more of a C/C++ construct. In this case it is more typical to have the method create the ArrayList itself and then return that instead of void